source/heroes/ScreensaverHero.hx

```diff
@@ -5,43 +5,67 @@ import flixel.util.FlxColor;
 import flixel.FlxG;
 
 class ScreensaverHero extends FlxSprite {
-    var DirectionX:Float;
-    var DirectionY:Float;
+    var HeroHeight:Int = 50;
+    var HeroWidth:Int = 25;
+    var StartingHeroColor:FlxColor = 0xFFFFFFFF;
+    var Blue:FlxColor = 0xFF0000FF;
+    var Yellow:FlxColor = 0xFFFFFF00;
+    var Green:FlxColor = 0xFF008000;
+    var Red:FlxColor = 0xFFFF0000;
+    var SpeedX:Float = FlxG.random.float(-50, 50);
+    var SpeedY:Float = FlxG.random.float(-50, 50);
 
     public function new(X:Float=0, Y:Float=0)
     {
         super(X, Y);
-        makeGraphic(20, 40, 0xFFFFFFFF);
-        this.DirectionX = FlxG.random.float(0,1);
-        this.DirectionY = FlxG.random.float(0,1);
+        super.makeGraphic(HeroWidth, HeroHeight, StartingHeroColor);
+        this.velocity.x = SpeedX;
+        this.velocity.y = SpeedY;
     }
     override public function update(elapsed:Float)
     {
-        super.update(elapsed);
-        if(this.DirectionX < .5){
-            this.x += 1;
+        if (this.velocity.y > 0 && this.y >= FlxG.height){
+            this.y = 0; 
+        }
+        else if (this.velocity.y < 0 && (this.y + this.height) <= 0){
+            this.y = FlxG.height;
+        }
+
+        if (this.velocity.x > 0 && this.x >= FlxG.width){
+            this.x = -(this.width);   
         }
-        else{
-            this.x -= 1;
+        else if (this.velocity.x < 0 && (this.x + this.width) <= 0){
+            this.x = FlxG.width; 
         }
-        if(this.DirectionY < .5){
-            this.y += 1;
+        // Accelerate screensaver heroes if spacebar is pressed
+        if(FlxG.keys.pressed.SPACE)
+        {
+            SpeedX *= 1.01;
+            SpeedY *= 1.01;
         }
-        else{
-            this.y -= 1;
+        // Reinitialize speeds if R is pressed
+        if(FlxG.keys.pressed.R)
+        {
+            SpeedX = FlxG.random.float(-50, 50);
+            SpeedY = FlxG.random.float(-50, 50);
         }
 
         if((this.x) > 150 ){
-            this.color = 0xFF00FF00;
+            this.color = Green;
         }
         if((this.y) > 250 ){
-            this.color = 0xFFFA0678;
+            this.color = Red;
         }
         if((this.x) < 50 ){
-            this.color = 0xFF4400FF;
+            this.color = Blue;
         }
         if((this.y) < 50 ){
-            this.color = 0xFFF7E602;
+            this.color = Yellow;
         }
+
+        this.velocity.x = SpeedX;
+        this.velocity.y = SpeedY;
```

Comment:
If the only purpose of SpeedX and SpeedY is to be set to `velocity.x` and `velocity.y`, I think they're not terribly useful variables.

I think they might have come out of our effort to reduce "magic number" usage, but I don't think they're actually making the code any easier to read or work with as-is. It might even make us more bug-prone if we forget about it later and start mixing direct velocity updates with this `SpeedX` and `SpeedY` stuff.

I think it would be cleaner to always act on velocity directly - to keep with the theme of cutting out magic numbers, the lines at the top:
```haxe
    var SpeedX:Float = FlxG.random.float(-50, 50);
    var SpeedY:Float = FlxG.random.float(-50, 50);
```
could instead be something like 
```haxe
    var maxVelocityMagnitude = 50;
```
which can be used with random number generation to get your starting values.

No changes required, work at it at your own discretion.